hours/tests/test_api_permissions.py

```diff
@@ -0,0 +1,972 @@
+import json
+
+import pytest
+from django.core.serializers.json import DjangoJSONEncoder
+from django.urls import reverse
+
+from hours.models import (
+    DatePeriod,
+    Resource,
+    Rule,
+    TimeSpan,
+    _get_all_parent_organizations,
+)
+
+
+#
+# Resource
+#
+@pytest.mark.django_db
+def test_create_resource_anonymous(api_client):
+    url = reverse("resource-list")
+
+    data = {"name": "Test name"}
+
+    response = api_client.post(
+        url,
+        data=json.dumps(data, cls=DjangoJSONEncoder),
+        content_type="application/json",
+    )
+
+    assert response.status_code == 403, "{} {}".format(
+        response.status_code, response.data
+    )
+
+
+@pytest.mark.django_db
+def test_create_resource_authenticated_no_org(user, api_client):
+    api_client.force_authenticate(user=user)
+
+    url = reverse("resource-list")
+
+    data = {"name": "Test name"}
+
+    response = api_client.post(
+        url,
+        data=json.dumps(data, cls=DjangoJSONEncoder),
+        content_type="application/json",
+    )
+
+    assert response.status_code == 400, "{} {}".format(
+        response.status_code, response.data
+    )
+
+
+@pytest.mark.django_db
+def test_create_resource_authenticated_has_org(
+    organization_factory, data_source, user, api_client
+):
+    organization = organization_factory(
+        origin_id=12345,
+        data_source=data_source,
+        name="Test organization",
+    )
+
+    organization.regular_users.add(user)
+    api_client.force_authenticate(user=user)
+
+    url = reverse("resource-list")
+
+    data = {
+        "name": "Test name",
+        "organization": organization.id,
+    }
+
+    response = api_client.post(
+        url,
+        data=json.dumps(data, cls=DjangoJSONEncoder),
+        content_type="application/json",
+    )
+
+    assert response.status_code == 201, "{} {}".format(
+        response.status_code, response.data
+    )
+
```

Comment:
Should we add a test (and some logic) about creating children to resources depending on organization?

I'm not sure about this, but I believe TPREK doesn't want anybody adding subsections as child resources if they don't have edit rights to the original resource.

Therefore, if a new resource has parent, the parent's organization should determine the create rights. And if an existing resource has parents, then _get_all_parent_organizations should determine edit rights.

Yeah, there are some issues here, because a subsection might have lots of parents, and in some case they might have different organizations. Then being an admin in *one* organization would allow you to edit the subsection under *all* resources.

Reply:
True. Didn't think about that. Added an organization check when adding a child resource. For now it only checks the immediate parents.